[2023-03-16T05:13:38.976+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: Spotify_Data_ETL.ValidateAndLoadArtistDim.ValidateArtistDim manual__2023-03-16T05:02:48.467376+00:00 [queued]>
[2023-03-16T05:13:38.994+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: Spotify_Data_ETL.ValidateAndLoadArtistDim.ValidateArtistDim manual__2023-03-16T05:02:48.467376+00:00 [queued]>
[2023-03-16T05:13:38.998+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2023-03-16T05:13:39.000+0000] {taskinstance.py:1280} INFO - Starting attempt 13 of 15
[2023-03-16T05:13:39.002+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2023-03-16T05:13:39.019+0000] {taskinstance.py:1300} INFO - Executing <Task(PythonOperator): ValidateAndLoadArtistDim.ValidateArtistDim> on 2023-03-16 05:02:48.467376+00:00
[2023-03-16T05:13:39.027+0000] {standard_task_runner.py:55} INFO - Started process 9497 to run task
[2023-03-16T05:13:39.041+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'Spotify_Data_ETL', 'ValidateAndLoadArtistDim.ValidateArtistDim', 'manual__2023-03-16T05:02:48.467376+00:00', '--job-id', '1236', '--raw', '--subdir', 'DAGS_FOLDER/spotifydag.py', '--cfg-path', '/tmp/tmp2i0emivi']
[2023-03-16T05:13:39.045+0000] {standard_task_runner.py:83} INFO - Job 1236: Subtask ValidateAndLoadArtistDim.ValidateArtistDim
[2023-03-16T05:13:39.200+0000] {task_command.py:388} INFO - Running <TaskInstance: Spotify_Data_ETL.ValidateAndLoadArtistDim.ValidateArtistDim manual__2023-03-16T05:02:48.467376+00:00 [running]> on host 18c42e4ef76b
[2023-03-16T05:13:39.827+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=Ananta Moharana
AIRFLOW_CTX_DAG_ID=Spotify_Data_ETL
AIRFLOW_CTX_TASK_ID=ValidateAndLoadArtistDim.ValidateArtistDim
AIRFLOW_CTX_EXECUTION_DATE=2023-03-16T05:02:48.467376+00:00
AIRFLOW_CTX_TRY_NUMBER=13
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-03-16T05:02:48.467376+00:00
[2023-03-16T05:13:39.829+0000] {logging_mixin.py:137} INFO - None
[2023-03-16T05:13:39.829+0000] {logging_mixin.py:137} INFO - ['artist_name', 'id']
[2023-03-16T05:13:39.830+0000] {logging_mixin.py:137} INFO - artist_dimension
[2023-03-16T05:13:39.830+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/src/transform.py", line 74, in validate_data
    data_table=pd.read_json(data)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 207, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 614, in read_json
    return json_reader.read()
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 748, in read
    obj = self._get_object_parser(self.data)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 770, in _get_object_parser
    obj = FrameParser(json, **kwargs).parse()
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 885, in parse
    self._parse_no_numpy()
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 1140, in _parse_no_numpy
    loads(json, precise_float=self.precise_float), dtype=None
ValueError: Unexpected character found when decoding 'NaN'
[2023-03-16T05:13:39.852+0000] {taskinstance.py:1323} INFO - Marking task as UP_FOR_RETRY. dag_id=Spotify_Data_ETL, task_id=ValidateAndLoadArtistDim.ValidateArtistDim, execution_date=20230316T050248, start_date=20230316T051338, end_date=20230316T051339
[2023-03-16T05:13:39.868+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 1236 for task ValidateAndLoadArtistDim.ValidateArtistDim (Unexpected character found when decoding 'NaN'; 9497)
[2023-03-16T05:13:39.939+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2023-03-16T05:13:39.997+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
